### ISSUE TITLE
Including Headings in Wikilinks

### ISSUE THREAD

**Issue by ChristinWhite** [foambubble/foam#536]
- Foam version: v0.11.0
- Platform: Mac
- Issue occur on the [foam template](https://github.com/foambubble/foam-template) repo: Yes

**Summary**
Wikilinks don't behave properly when linking to a heading within a file using the syntax `[[note-name#heading]]`. 

Specifically, Foam wants to treat them as entirely new notes on the graph and in placeholders. If I go to _the destination note_ and look at backlinks _the origin note_ does not show up.

**Steps to reproduce**
1. Create a wikilink to an existing note and select a heading. For example I added the following to inbox.md in the template.

```markdown
[[daily-notes#Configuration]] 
[[daily-notes#Shortcuts and Snippets]]
```

2. Check placeholders and graph in `inbox.md`
3. Check backlinks in `daily-notes.md`
4. If you add `[Daily Notes](docs/features/daily-notes.md#Configuration)` to `inbox.md` the graph shows the link to `daily-notes`, a placeholder isn't created and a backlink shows up in `daily-notes.md` as expected.

**Additional information**
- Ideally, it would be great if wiki links supported headings within the note itself, `[[#Configuration]]` for navigating the note manually for compatibility in other applications that don't have an outline feature. Obsidian sort of supports relative heading links but wants to explicitly add the full filename if you change the filename.
- [Foam log](https://gist.github.com/ChristinWhite/73f8f7a53baf54a6f62670f07c184954) <sup>[1](#footnote)</sup>

**Screenshots**
Step 1:
<img width="669" alt="CleanShot 2021-03-17 at 11 46 57@2x" src="https://user-images.githubusercontent.com/1746405/111496227-824c4980-8716-11eb-8bea-125c51612820.png">
Step 2:
<img width="516" alt="CleanShot 2021-03-17 at 11 45 30@2x" src="https://user-images.githubusercontent.com/1746405/111496030-4ca76080-8716-11eb-8620-de0452855fb5.png">
<img width="504" alt="CleanShot 2021-03-17 at 11 45 52@2x" src="https://user-images.githubusercontent.com/1746405/111496081-59c44f80-8716-11eb-9159-854bbbc145aa.png">
Step 3:
<img width="449" alt="CleanShot 2021-03-17 at 11 46 34@2x" src="https://user-images.githubusercontent.com/1746405/111496166-719bd380-8716-11eb-9f3e-8b797bea66ce.png">

What this looks like in the context of my own notes:
![image](https://user-images.githubusercontent.com/1746405/111501627-862e9a80-871b-11eb-9853-986e4c8c70fe.png)

**How standard is this? Not standard but supported in Bear and Obsidian**
- ❌ [Markdown Notes](https://marketplace.visualstudio.com/items?itemName=kortina.vscode-markdown-notes) also treats this improperly and if `createNoteOnGoToDefinitionWhenMissing` is enabled it will create new notes including the heading text (`daily-notesconfiguration.md` and `daily-notesshortcuts-and-snippets` ).
- ❌ [Markdown Links](https://marketplace.visualstudio.com/items?itemName=tchayen.markdown-links) does not show a relationship to `daily-notes.md` and since it doesn't show placeholders it's likely just ignoring those links entirely.
- ✔️ [Obsidian](https://obsidian.md) supports headings in wikilinks, as does [Bear](https://bear.app).
- ❌ [Noteplan](https://noteplan.co), [DEVONthink](https://www.devontechnologies.com/apps/devonthink) and [Drafts](https://getdrafts.com) do not.

<a name="footnote">1</a>: The Foam documentation to `foam-logging-in-vscode.md` in the issue template seems to be broken.

**Comment by riccardoferretti** [comment]
related to #512

**Comment by riccardoferretti** [comment]
this will be addressed by #856

**Comment by riccardoferretti** [comment]
Released in 0.17.0
